src/states/finitemps.jl

```diff
@@ -282,6 +282,35 @@ end
 
 Base.checkbounds(::Type{Bool}, ψ::FiniteMPS, i::Integer) = 1 <= i <= length(ψ)
 
+Base.@propagate_inbounds function Base.getindex(ψ::FiniteMPS, i::Int)
+    c = ψ.center
+
+    @boundscheck begin
+        (1 <= i <= length(ψ)) || throw(BoundsError(ψ, i))
+    end
+
+    if isinteger(c)
+        return if i > Int(c)
+            ψ.AR[i]
```

Comment:
Because you are checking the bounds above, in principle you should be allowed to put an `@inbounds` here. Honestly, I wouldn't bother with it, since we are not guaranteed that these arrays are the actual sizes they should be, in which case this would lead to segfaults. This should never really be called in super tight loops, so this is fine as is.